ssl/statem/statem_srvr.c

```diff
@@ -3527,13 +3631,22 @@ MSG_PROCESS_RETURN tls_process_client_certificate(SSL_CONNECTION *s,
     SSL_SESSION *new_sess = NULL;
     SSL_CTX *sctx = SSL_CONNECTION_GET_CTX(s);
 
+    if (s->ext.client_cert_type == TLSEXT_cert_type_rpk)
+        return tls_process_client_rpk(s, pkt);
+
     /*
      * To get this far we must have read encrypted data from the client. We no
      * longer tolerate unencrypted alerts. This is ignored if less than TLSv1.3
      */
     if (s->rlayer.rrlmethod->set_plain_alerts != NULL)
         s->rlayer.rrlmethod->set_plain_alerts(s->rlayer.rrl, 0);
```

Comment:
This code is duplicated at the start of `tls_process_client_rpk`. Why not move it above the call to `tls_process_client_rpk` above and have it in just one place?